### PR TITLE
Refactor files in test/ so they can be run independently

### PR DESCRIPTION
**Pull request by odow**
- Make it so individual test files can be run
- Time which tests take the most time
  - Currently, MutableArithmetics is the worst offender
- Remove unneeded test entries in Project.toml

Refactored out of #2277

### CODE REVIEW

File: test/derivatives.jl [hunk 1]
```diff
@@ -345,31 +343,34 @@ function dualforward(ex, x; ignore_nan=false)
...
 
     # compare with running dual numbers
-    forward_dual_storage = zeros(DualNumbers.Dual{Float64},length(nd))
-    partials_dual_storage = zeros(DualNumbers.Dual{Float64},length(nd))
-    output_dual_storage = zeros(DualNumbers.Dual{Float64},length(x))
-    reverse_dual_storage = zeros(DualNumbers.Dual{Float64},length(nd))
-    x_dual = [DualNumbers.Dual(x[i],1.0) for i in 1:length(x)]
+    _epsilon(x::ForwardDiff.Dual{Nothing, Float64, 1}) = x.partials[1]
+
+    forward_dual_storage = zeros(ForwardDiff.Dual{Nothing, Float64, 1},length(nd))
```
Comment (blegat) [inline test/derivatives.jl:348]: Great, it was always complaining that I didn't have `DualNumbers` because I was not running the tests with `] test`.

**Comment by odow** [comment]
Objections to merging? Once this is in I can start teeing up the rest of the Test PRs.

File: test/constraint.jl [hunk 2]
```diff
@@ -1,3 +1,13 @@
+using JuMP
+using LinearAlgebra
+using Test
+
+include(joinpath(@__DIR__, "utilities.jl"))
+
+@static if !(:JuMPExtension in names(Main))
```
Comment (mlubin) [inline test/constraint.jl:7]: Why do we do this unusual check for JuMPExtension.jl but not utilities.jl?

Comment (odow) [reply test/constraint.jl:7]: Utilities just loads some functions, whereas JuMPExtension is a module so we get annoying `WARNING: module JuMPExtension has been redefined` warnings.

The other PRs fix this by making each file a module. The module approach also prevents state leaking between files, and allows us to programmatically determine the tests to run within a particular file.

**Pull request merged**